src/drivers/gitlab.js

```diff
@@ -144,28 +144,40 @@ class Gitlab {
     return { uri: `${repo}${url}`, mime, size };
   }
 
-  async runnerToken() {
+  async runnerToken(body) {
     const projectPath = await this.projectPath();
-    const endpoint = `/projects/${projectPath}`;
+    const legacyEndpoint = `/projects/${projectPath}`;
+    const endpoint = `/user/runners`;
 
-    const { runners_token: runnersToken } = await this.request({ endpoint });
+    const { id, runners_token: runnersToken } = await this.request({
+      endpoint: legacyEndpoint
+    });
+
+    if (runnersToken === null) {
+      if (!body) body = new URLSearchParams();
+      body.append('project_id', id);
+      body.append('runner_type', 'project_type');
+      return (await this.request({ endpoint, method: 'POST', body })).token;
+    }
```

Comment:
GitLab decided to change the way of registering runners, breaking the old approach; now we support both for good measure, and "infer [sic] which to use automatically" like in the good ol' CML times. 🙃 

https://docs.gitlab.com/ee/ci/runners/new_creation_workflow.html